src/test/ui/parser/trait-object-lifetime-parens.rs

```diff
@@ -8,7 +8,8 @@ fn check<'a>() {
     let _: Box<Trait + ('a)>; //~ ERROR parenthesized lifetime bounds are not supported
     let _: Box<('a) + Trait>;
     //~^ ERROR expected type, found `'a`
-    //~| ERROR expected `:`, found `)`
+    //~| ERROR expected `while`, `for`, `loop` or `{` after a label
+    //~| ERROR expected expression, found `)`
```

Comment:
Should we maybe bail earlier on failures in path context? We're "recovering", but that only causes further errors to accumulate.

Reply:
I'll investigate what exactly leads us into this recovery path and why we're suddenly trying to parse an expression here.

Reply:
The root cause was the backtracking in `parse_local`. I've solved the issue by making the type parser recover when there's a `'lifetime` beginning a bare trait object when no `+` follows.